content/docs/capabilities/mcp.mdx

```diff
@@ -0,0 +1,727 @@
+    to: http://internal-server:8080/mcp
+    mcp:
+      server: {}
+    policy:
+      allow:
+        and:
+          - domain:
+              is: trusted-domain.com
+          - groups:
+              has: mcp-users
+```
+
+### Token Security
+
+- External tokens (TE) have limited scope and lifetime
+- Internal tokens (TI) are never exposed to external clients
+- All tokens are automatically refreshed by Pomerium
+- Failed authentication attempts are logged for audit
+
+### Network Security
+
+- MCP servers should run on internal networks
+- Use TLS for all external communications
+- Implement proper firewall rules
+- Consider IP allowlisting for sensitive servers
+
+## Common Integration Patterns
+
+### Database Access
+
+```yaml
+routes:
+  - from: https://db-mcp.domain.com
+    to: http://postgres-mcp-server:8080/mcp
+    name: Company Database
+    mcp:
+      server:
+        max_request_bytes: 2097152 # 2MB for larger queries
+    policy:
+      allow:
+        and:
+          - domain:
+              is: company.com
+          - groups:
+              has: data-analysts
+```
+
+### GitHub Integration
+
+```yaml
+routes:
+  - from: https://github-mcp.domain.com
+    to: http://github-mcp-server:8080/mcp
+    name: GitHub Repository Access
+    mcp:
+      server:
+        upstream_oauth2:
+          client_id: 'github-oauth-client-id'
+          client_secret: 'github-oauth-client-secret'
+          endpoint:
+            auth_url: 'https://github.com/login/oauth/authorize'
+            token_url: 'https://github.com/login/oauth/access_token'
+          scopes: ['read:user', 'repo', 'user:email']
+```
+
+### Notion Integration
+
+```yaml
+routes:
+  - from: https://notion-mcp.domain.com
+    to: http://notion-mcp-server:8080/mcp
+    name: Notion Workspace
+    mcp:
+      server:
+        upstream_oauth2:
+          client_id: 'notion-oauth-client-id'
+          client_secret: 'notion-oauth-client-secret'
+          endpoint:
+            auth_url: 'https://api.notion.com/v1/oauth/authorize'
+            token_url: 'https://api.notion.com/v1/oauth/token'
+          scopes: ['read_content', 'read_user_with_email']
+```
+
+## Troubleshooting
+
+### Common Issues
+
+**Server shows as disconnected:**
+
+- Check if upstream OAuth is configured correctly
+- Verify OAuth client credentials
+- Ensure user has completed authentication flow
+- Check Pomerium logs for OAuth errors
+
+**Authentication failures:**
+
+- Verify route policies allow the user
+- Check token expiration and refresh logic
+- Ensure external tokens are properly formatted
+- Review access logs for policy violations
+
+**Request size errors:**
+
+- Increase `max_request_bytes` in server configuration
+- Check for oversized request payloads
+- Consider request compression if supported
+
+### Logging and Monitoring
+
+Enable detailed logging for MCP operations:
+
+```yaml
+log_level: info
+# Additional logging options in global config
+```
+
+Monitor key metrics:
```

Comment:
Can we provide Pomerium metrics to watch these? How would someone monitor these?